realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -172,6 +172,11 @@ static synchronized void release(BaseRealm realm) {
             return;
         }
 
+        realm.tidyPerformedTransactions();
+        if (!realm.asyncTransactions.isEmpty() || !realm.asyncTransactionCallbacks.isEmpty()) {
```

Comment:
I moved the list to RealmThreadPoolExecutor and the checking routine to `BaseRealm.doClose()`.